### PR TITLE
Implement a UI for editing approval configs

### PR DESCRIPTION
**Pull request by jrobbins**
This is progress on the requests that API owners made a couple of weeks ago to be able to set a next-action date, an owner, and specify that certain origin trials require 3 LGTMs.

In this PR:
* In chromedash-approvals-dialog: Add new fetching, rendering, and input handling logic for the new fields.  The fields are initially not shown but are shown if they have a value or if the user clicks an expansion icon.
* Change the API and tests to treat all three fields (owner, nextAction, and additionalReview) as required.  That is so that the user can clear each field by providing an empty value.  Also, add a list of possible owners to choose from.
* Tweak cs-client to provide empty values of the right type.
* In approval_devs.py: Add a missing reference to the approval for extending an experiment.

### CODE REVIEW

**Comment by jrobbins** [comment]
Here's a screenshot:
image

File: static/elements/chromedash-approvals-dialog.js [hunk 1]
```diff
@@ -405,6 +523,16 @@ class ChromedashApprovalsDialog extends LitElement {
   handleCancel() {
     this.shadowRoot.querySelector('chromedash-dialog').close();
   }
+
+  toggleConfig(approvalDef) {
+    let newConfigs = new Set([...this.showConfigs]); // Make a copy.
```
Comment (KyleJu) [inline static/elements/chromedash-approvals-dialog.js:528]: Why are we making a copy? Is it to trigger certain functions?

File: static/elements/chromedash-approvals-dialog.js [hunk 2]
```diff
@@ -230,6 +251,63 @@ class ChromedashApprovalsDialog extends LitElement {
...
+    const additionalReviewWidget = html`
+      <input id="additional-review-${fieldId}" data-field="${fieldId}"
+             @change=${this.handleConfigChanged}
+             type=checkbox ?checked=${additionalReview}>
+    `;
+
+    return html`
+     <table class="config-area">
+       <tr>
+         <td>Owner:</td>
```
Comment (KyleJu) [inline static/elements/chromedash-approvals-dialog.js:294]: Owners?

File: static/elements/chromedash-approvals-dialog.js [hunk 3]
```diff
@@ -159,7 +173,14 @@ class ChromedashApprovalsDialog extends LitElement {
...
       });
-    Promise.all([p1, p2, p3]).then(() => {
+    const p4 = window.csClient.getApprovalConfigs(this.featureId).then(
+      (res) => {
+        this.configs = res.configs;
+        this.showConfigs = new Set(this.configs.map(c => c.field_id));
+        this.changedConfigsByField = new Map();
+        this.possibleOwners = res.possible_owners;
+      });
+    Promise.all([p1, p2, p3, p4]).then(() => {
```
Comment (KyleJu) [inline static/elements/chromedash-approvals-dialog.js:183]: IIUC, when `p1,p2,p3,p4` promises have resolved, this is resolved. What if one of them get rejected, is `this.loading` still true?

File: static/elements/chromedash-approvals-dialog.js (hunk 1 above)
Comment (jrobbins) [reply static/elements/chromedash-approvals-dialog.js:528]: Yes, it is to trigger a new call to render(). 

My understanding is that modifying a collection is not detected as a change to a lit-element property.  Instead, we need to set the property itself to point to a new collection.  This page is not explicit about it, but it says that the first step in the process is that `the property setter is called`.  That is not going to happen if we simply mutate an existing property value.
https://lit.dev/docs/v1/components/properties/#what-happens-when-properties-change

File: static/elements/chromedash-approvals-dialog.js (hunk 2 above)
Comment (jrobbins) [reply static/elements/chromedash-approvals-dialog.js:294]: Changed.

File: static/elements/chromedash-approvals-dialog.js (hunk 3 above)
Comment (jrobbins) [reply static/elements/chromedash-approvals-dialog.js:183]: Yes, that is true.  If any one of the individual promises is rejected, then the Promise.all() also rejects.  There is no reject case, so the user would just sit there seeing a spinner.

To improve that UX, I have added a catch() clause that shows an error message and closes the dialog.

File: static/elements/chromedash-approvals-dialog.js [hunk 4]
```diff
@@ -230,6 +255,63 @@ class ChromedashApprovalsDialog extends LitElement {
...
 
+  renderConfigWidgets(approvalDef) {
+    const fieldId = approvalDef.id;
+    const isOpen = this.showConfigs.has(fieldId);
+    if (!isOpen) {
+      return nothing;
+    }
+
+    const config = this.configs.find(c => c.field_id == fieldId);
+    const owners = (config && config.owners || []).join(', ');
```
Comment (KyleJu) [inline static/elements/chromedash-approvals-dialog.js:266]: This line is not very intuitive. I understand it used as non-Boolean operands but according to the API `[a5 = 'Cat' && 'Dog'      // t && t returns "Dog"]`. It is not clear what precedence is here.

File: static/elements/chromedash-approvals-dialog.js (hunk 3 above)
Comment (KyleJu) [reply static/elements/chromedash-approvals-dialog.js:183]: Yup thanks for addressing it

File: static/elements/chromedash-approvals-dialog.js [hunk 5]
```diff
@@ -230,6 +255,63 @@ class ChromedashApprovalsDialog extends LitElement {
...
+    const nextActionWidget = html`
+      <input id="next-action-${fieldId}" data-field="${fieldId}"
+             @change=${this.handleConfigChanged}
+             type=date name="next_action_${fieldId}"
+             value="${nextAction}">
+    `;
+    const additionalReviewWidget = html`
+      <input id="additional-review-${fieldId}" data-field="${fieldId}"
+             @change=${this.handleConfigChanged}
+             type=checkbox ?checked=${additionalReview}>
```
Comment (KyleJu) [inline static/elements/chromedash-approvals-dialog.js:292]:
```suggestion
             type=checkbox ? checked=${additionalReview}>
```

**Review by KyleJu — APPROVED** [review]
LGMT

File: static/elements/chromedash-approvals-dialog.js (hunk 1 above)
Comment (KyleJu) [reply static/elements/chromedash-approvals-dialog.js:532]: `That is not going to happen if we simply mutate an existing property value.` 
> Yup that is the case for Polymer as well. Thanks for the explanation

File: static/elements/chromedash-approvals-dialog.js (hunk 5 above)
Comment (jrobbins) [reply static/elements/chromedash-approvals-dialog.js:292]: The leading `?` controls whether the attribute is present or not.  It is normally written without a space.
https://lit.dev/docs/v1/components/templates/#bind-properties-to-templated-elements

Comment (KyleJu) [reply static/elements/chromedash-approvals-dialog.js:292]: Aha I see.

File: static/elements/chromedash-approvals-dialog.js (hunk 4 above)
Comment (jrobbins) [reply static/elements/chromedash-approvals-dialog.js]: I rewrote these as conditional expressions (like `c ? x : y`) and I think that that is an improvement.  Thanks.

**Pull request merged**